src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +32,16 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        this.value = processInputAddress(trimmedAddress);
+    }
+
+    private String processInputAddress(String inputAddress) {
+        String[] splitAddress = inputAddress.split(ADDRESS_SEPARATOR);
+        Block block = new Block(splitAddress[ADDRESS_BLOCK]);
```

Comment:
Aren't these variables scoped within this method alone?
How are you storing these values? 
I notice that you still retain the `value` variable as it is.